tests/flox-init.bats

```diff
@@ -0,0 +1,232 @@
+#! /usr/bin/env bats
+# -*- mode: bats; -*-
+# ============================================================================ #
+#
+# Test `flox init' sub-command.
+#
+# This sub-command's `--template' option takes an "installable" URI as its
+# argument, and we focus on auditing various URI edge cases here.
+#
+# NOTE: the URIs accepted by `flox init --template <URI>' do not accept an
+# extended output spec ( `^out,bin,dev' suffix ).
+#
+#
+# ---------------------------------------------------------------------------- #
+
+load test_support.bash;
+
+# bats file_tags=uri, init, uri:project
+
+
+# ---------------------------------------------------------------------------- #
+
+setup_file() {
+  common_file_setup;
+
+  # Suppresses warning messages that clutter backtraces.
+  git config --global init.defaultBranch main;
+
+  export _floxpkgs_rev="2c75b96bc3e8c78b516b1fc44dbf95deae6affca";
+
+  # Create an alias for testing indirects
+  $FLOX_CLI nix registry add floxpkgs-alias github:flox/floxpkgs;
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+# Helpers for project based tests.
+
+project_setup() {
+  export PROJECT_DIR="${BATS_TEST_TMPDIR?}/project-${BATS_TEST_NUMBER?}";
+  rm -rf "$PROJECT_DIR";
+  mkdir -p "$PROJECT_DIR";
+  pushd "$PROJECT_DIR" >/dev/null||return;
+  git init;
+}
+
+project_teardown() {
+  popd >/dev/null||return;
+  rm -rf "${PROJECT_DIR?}";
+  unset PROJECT_DIR;
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+setup()    { common_test_setup; project_setup;       }
+teardown() { project_teardown; common_test_teardown; }
+
+
+# ---------------------------------------------------------------------------- #
+
+# bats test_tags=uri:github
+@test "'flox init -t github:flox/floxpkgs#project'" {
+  run "$FLOX_CLI" init -n "${PWD##*/}" -t 'github:flox/floxpkgs#project';
```

Comment:
Is `"${PWD##*/}"` removing everything except the final path component?

Some usages of parameter expansion could use clarification for people that aren't Bash wizards. Something as simple as:
```
local dir_name="${PWD##*/}"
```
would make things clearer

Reply:
You're right that the expansion `${PWD##*/}` is equivalent to `$( basename $( pwd; ); )` or `$( basename "$PWD"; )`.

I'm writing up some other tickets right now but if you use the "suggest a patch" I can commit it, or go back to add temporary variables later.

This might be a useful reference for the couple of expansions you asked about in this file: https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html

Reply:
I actually had that up while I was reading this :)

Reply:
should be good now.